pkg/service/tap_service.go

```diff
@@ -65,9 +78,19 @@ func (tapService *TAPService) ConnectToTurbo() {
 	tapService.disconnectFromTurbo = make(chan struct{})
 	isRegistered := make(chan bool, 1)
 	defer close(isRegistered)
-
+	// Get the credential from the secret
```

Comment:
I dont see any backward compatibility handled in here or almost anywhere. 
I mean, tracing the flow it seems to necessarily fail if I don't create the secret or provide the client `id` and `secret`.
Am I missing something?

Reply:
Thanks for explaining.